omnigibson/prims/rigid_kinematic_prim.py

```
@@ -0,0 +1,235 @@
+    A kinematic-only object is not subject to simulator dynamics, and remains fixed unless the user
+    explicitly sets the body's pose / velocities.
+
+    Args:
+        relative_prim_path (str): Scene-local prim path of the Prim to encapsulate or create.
+        name (str): Name for the object. Names need to be unique per scene.
+        load_config (None or dict): If specified, should contain keyword-mapped values that are relevant for
+            loading this prim at runtime.
+    """
+
+    def __init__(
+        self,
+        relative_prim_path,
+        name,
+        load_config=None,
+    ):
+        # Caches for kinematic-only objects
+        # This exists because RigidPrimView uses USD pose read, which is very slow
+        self._kinematic_world_pose_cache = None
+
+        # Run super init
+        super().__init__(
+            relative_prim_path=relative_prim_path,
+            name=name,
+            load_config=load_config,
+        )
+
+    def _post_load(self):
+        # Make sure it's set to be kinematic
+        if not self.is_attribute_valid("physics:kinematicEnabled"):
+            self.create_attribute("physics:kinematicEnabled", True)
+        if not self.is_attribute_valid("physics:rigidBodyEnabled"):
+            self.create_attribute("physics:rigidBodyEnabled", False)
+        self.set_attribute("physics:kinematicEnabled", True)
+        self.set_attribute("physics:rigidBodyEnabled", False)
+
+        # Run super method to handle common functionality
+        super()._post_load()
+
+    def update_handles(self):
+        """
+        Kinematic objects don't need to update handles as they don't use RigidPrimView
+        """
+        pass
+
+    def set_linear_velocity(self, velocity):
+        """
+        Kinematic objects don't have dynamic velocities - this is a no-op
+
+        Args:
+            velocity (th.tensor): linear velocity to set (ignored). Shape (3,).
+        """
+        pass
```

should we raise an error here?